File: data-uploader.js
/**
 * Module dependencies.
 */
const request = require('request');
const readline = require('readline');
const fs = require('fs');

const CSV = '../rbc_sectran.csv';

var i = 0;

var pooledRequest = request.defaults({pool: {maxSockets: 5}});

var lineReader = require('readline').createInterface({
 crlfDelay: 1000,
 input: fs.createReadStream(CSV)
});

lineReader.on('line', function (line) {

  var jsonLine = { "csvLine": line };

//  pooledRequest.post('https://eagles-app.mybluemix.net/api/transactions',
  pooledRequest.post('http://localhost:3000/api/transactions',
  { json: jsonLine },
      function (error, response, body) {

          if (!error && response.statusCode == 200) {
              console.log('uploaded ', ++i);
          } else {
            console.log('error: ', error);
            console.log('response: ', response);
            console.log('body: ', body);
          }
      }
  );


});

File: data-uploader.py
#!/usr/bin/python

import requests

dataFile="../rbc_sectran.csv"
#url="http://localhost:3000/api/transactions"
url="https://eagles-app.mybluemix.net/api/transactions"

def postData( line ):
    try:
        requests.post(url, data={'csvLine': line})
    except ZeroDivisionError:
        print "*** failed ***"

i=0
with open(dataFile) as f:
    for line in f:
        postData(line)
        i=i+1
        print "Posted: " + str(i)

